src/UI/Canvas.cs

```diff
@@ -0,0 +1,162 @@
+using TileMapper.Windowing;
+using Raylib_cs;
+using ImGuiNET;
+using System.Numerics;
+
+namespace TileMapper {
+    
+    public class Canvas : GraphicsWindow {
+
+        int mapHeight, mapWidth;
+
+        int unitSize;
+
+        TileMap tMap;
+
+        TileLayer currentLayer;
+
+        TileSelector ts;
+
+        TileSet set = null;
+
+        int trueWidth, trueHeight;
+        float currentWidth, currentHeight;
+
+        // Scale = currernt/true.
+        float scaleX, scaleY; 
+
+        bool sizeSet = false;
+
+
+        // The width and height should be that of the tile map.
+        public Canvas(TileSelector ts, TileMap tMap) : base(0,0) {
+            
+            this.tMap = tMap;
+
+            mapWidth = tMap.GetRows();
+            mapHeight = tMap.GetCols();
+            unitSize = tMap.GetUnitWidth();
+            trueWidth = mapWidth*unitSize;
+            trueHeight = mapHeight*unitSize;
+
+            this.ResizeRenderTarget(trueWidth, trueHeight);
+
+            this.ts = ts;
+        }
+
+        public override void DrawUI() {
+
+            // Canvas is 600x600, other window elements are 8x27x8x8 found from GetWindowContentRegionMin.
```

Comment:
Is there a way to do this programatically? At the very least use constants defined somewhere to avoid "magic numbers".